### ISSUE TITLE
Radiation glue: cloud inhomogeneity clip, grey band mapping, and moisture conversions

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#678]
Found by a re-audit of the July 2026 ECHAM+RRTMGP review against `dev` @ `f6d1bcd`. Parent: #614.

Four residual items in the radiation glue, after #548 fixed the µ₀², surface albedo, pressure halo and condensate-as-vapour findings (all verified landed).

## 1. The 10 g/kg cap is a hard clip documented as an inhomogeneity scaling

`rrtmgp.py:55-60` and `:573-585`:
```python
_MAX_IN_CLOUD_CONDENSATE = 1.0e-2   # "The faithful-radiation equivalent of
                                    #  ECHAM's optics inhomogeneity factor"
... jnp.minimum(in_cloud_path(...), _MAX_IN_CLOUD_CONDENSATE)
```

ECHAM's `zinhoml` is a **continuous, LWP-dependent multiplicative rescaling applied to every cloudy cell**. `jnp.minimum(·, 1e-2)` is a hard one-sided clip that is the identity almost everywhere and flattens everything above the threshold to the same value — so deep convective cores and any thin-but-heavy cell become radiatively identical. It also has **zero gradient** above the threshold, killing `d(flux)/d(qc)` in exactly the optically thick cells that matter.

The comment is the more damaging half: a fidelity auditor reading it ticks off "sub-grid inhomogeneity: implemented" when no inhomogeneity scaling exists anywhere in the RRTMGP path.

Fix: replace the `minimum` with ECHAM's `zinhoml = min(1, (LWP/LWP_ref)^(−p))` applied multiplicatively (`mo_cloud_optics.f90`, `zinhoml`/`zinhomi` with the ECHAM defaults), so the cap becomes a smooth monotone reduction. Keep a very loose absolute ceiling purely as a NaN guard, and **rewrite both comments** to say what the code does.

## 2. Grey cloud-optics band mapping is broken

`cloud_optics.py:628-661` loops `range(N_SW_BANDS)` / `range(N_LW_BANDS)` — i.e. band ∈ {0,1} and {0,1,2} — into `get_band_wavelength` (`:41-87`), which holds a **6-entry** SW table and an **8-entry** LW table. Probe:

```
SW band true centers (um): [1.081, 0.310]
SW band table wavelength : [0.245, 0.305]
```

Band 0 — the near-IR band (0.69–2.5 µm) that carries essentially all cloud SW absorption — is evaluated at 0.245 µm where `n_imag = 1.1e-9`. For a 100 g/m² liquid layer: `ssa_sw = [0.99999, 0.99999]`, `tau_sw = [13.64, 13.64]` — identical in both bands, **zero absorption**. The 8-entry LW `k_abs` chains (`:462-483, 520-541`) are likewise indexed by the 3-band LW config, so the 500–2500 cm⁻¹ band (most of the OLR) gets a coefficient belonging to a different band set.

Compounding it, `radiation/constants.py:11-14` has the two SW labels **transposed**, in the file that declares itself the single source of truth:
```python
SW_BAND_LIMITS = (
    (4000, 14500),   # "UV + visible"   <- actually 690-2500 nm = near-IR
    (14500, 50000),  # "Near-IR"        <- actually 200-690 nm = UV+visible
)
```
Anyone reasoning about which band carries ozone/UV absorption vs water-vapour near-IR — or setting a per-band surface albedo, where visible and near-IR differ by a factor of several over snow and vegetation — picks the wrong index from these labels.

Reachability: no shipped Hydra config selects grey (`echam.yaml:35-39` deliberately blocks the CLI route), but `echam_terms.py:74` still has `radiation_scheme: str | PhysicsTerm = "grey"` as the **Python factory default**, and `grey_two_stream/radiation_scheme.py:392` calls `cloud_optics` in the live beam-split path. So a cheap grey A/B via `echam_physics(radiation_scheme="grey")` gets clouds that scatter SW but absorb none of it.

Fix: derive the representative wavelength from `SW_BAND_LIMITS`/`LW_BAND_LIMITS` directly (`λ = 1e4/(0.5·(wn_lo+wn_hi))` µm) instead of the hardcoded tables; replace the `jnp.where` `k_abs` chains with an `N_LW_BANDS`-length array indexed by band; swap the two comments in `constants.py` (**the numeric tuples' order is baked into downstream band indexing and must not move**). Add a test asserting `get_band_wavelength(b)` falls within band `b`'s own wavelength limits for every `b` — that single assertion catches all three sub-parts.

## 3. `q_t` handed to the library is a mixing ratio, so `1/(1−q)` is applied twice

`grey_two_stream/radiation_scheme.py:159` builds `h2o_vmr = q/(1−q)·1.608`, and `rrtmgp.py:307-308` converts back with `h2o_mass_mixing = h2o_vmr·c.eps`. Since `1.608·0.622 = 1.0002`, what reaches the library as `q_t` is the **mixing ratio** `r = q/(1−q)`, not `q`. The library then applies `mix_ratio = (q_t − q_c)/(1 − q_t)` — a second `1/(1−·)`.

Probe at q = 20 g/kg: true vmr 0.032811, library vmr 0.033500 — **+2.10 %**. Negligible above the boundary layer; a small systematic warm bias in tropical LW cooling and clear-sky SW absorption.

Fix: hand the library the specific humidity directly, inverting the mixing-ratio relation rather than passing `r`. Cleanest is to carry `specific_humidity` on `RadiationState` alongside `h2o_vmr` so the RRTMGP path never round-trips through the grey scheme's vmr convention. Test: pin the library's reconstructed vmr against `q/((1−q)·eps)` to 1e-6.

## 4. Upstream: dry `cp` in the flux→heating conversion

`jax-rrtmgp/rrtmgp/rte/two_stream.py:477` — `return constants.G * dflux / dp / constants.CP_D` — uses dry-air `cp` with no moisture correction, despite `CP_V = 1846.0` being defined two lines below. Correct is `cpd·(1 + 0.838·q)`, so heating rates are ~1.7 % too large at q = 20 g/kg in the tropical boundary layer.

**Fix belongs upstream:** a `jax-rrtmgp` PR threading the already-available `q_t`/`vmr_h2o` into `compute_heating_rate`. jcm needs no change beyond a version bump. Cross-link from `runs/UPSTREAM_ISSUE_jax-rrtmgp.md`.

## Checked and deliberately *not* filed

Aerosol optics are not delta-scaled while cloud optics are (`jax-rrtmgp/rrtmgp/rte/two_stream.py:385-393`). This **matches** reference RTE+RRTMGP practice, where `delta_scale()` is applied to cloud optics and the MERRA aerosol module does not delta-scale. Recorded as a known asymmetry; changing it would break agreement with the reference.

## Refs

Review findings 2.38e, 2.37, 2.38b, 2.38d. Related: #571, #537.
